StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1500CodeFixProvider.cs

```diff
@@ -229,28 +240,39 @@ private static bool ContainsStartOfLine(SyntaxToken token, int startLine)
 
         private static LinePosition GetTokenStartLinePosition(SyntaxToken token)
         {
-            return Location.Create(token.SyntaxTree, token.FullSpan).GetLineSpan().StartLinePosition;
+            return token.SyntaxTree.GetLineSpan(token.FullSpan).StartLinePosition;
         }
 
-        private class Rewriter : CSharpSyntaxRewriter
+        private static void AddReplacement(Dictionary<SyntaxToken, SyntaxToken> tokenReplacements, SyntaxToken originalToken, SyntaxToken replacementToken)
         {
-            private readonly Dictionary<SyntaxToken, SyntaxToken> tokenReplacements = new Dictionary<SyntaxToken, SyntaxToken>();
+            tokenReplacements[originalToken] = replacementToken;
+        }
 
-            public void AddReplacement(SyntaxToken originalToken, SyntaxToken replacementToken)
-            {
-                this.tokenReplacements.Add(originalToken, replacementToken);
-            }
+        private class FixAll : DocumentBasedFixAllProvider
+        {
+            public static FixAllProvider Instance { get; } =
+                new FixAll();
 
-            public override SyntaxToken VisitToken(SyntaxToken token)
-            {
-                SyntaxToken replacementToken;
+            protected override string CodeActionTitle =>
+                LayoutResources.SA1503CodeFix;
 
-                if (this.tokenReplacements.TryGetValue(token, out replacementToken))
+            protected override async Task<SyntaxNode> FixAllInDocumentAsync(FixAllContext fixAllContext, Document document)
+            {
+                var diagnostics = await fixAllContext.GetDocumentDiagnosticsAsync(document).ConfigureAwait(false);
+                if (diagnostics.IsEmpty)
                 {
-                    return replacementToken;
+                    return null;
                 }
 
-                return base.VisitToken(token);
+                SyntaxNode syntaxRoot = await document.GetSyntaxRootAsync().ConfigureAwait(false);
+
+                var tokens = diagnostics
+                    .Select(diagnostic => syntaxRoot.FindToken(diagnostic.Location.SourceSpan.Start))
+                    .OrderBy(token => token.SpanStart);
```

Comment:
The current construction is fine in this context. I would prefer to see `ToImmutableArray` included in the assignment to `tokens` instead of at the usage location.